### DISCUSSION TITLE
How to prevent dot (`.`) in number input?

### DISCUSSION THREAD

**Discussion by taviks** [alpinejs/alpine#946] [1. Help]
I have a number input that I was to only accept integers (no floats). Is there a recommended way of achieving this with Alpine?

These approaches **do not work**:

```html
x-on:keydown..="e.preventDefault()"
x-on:keydown.period="e.preventDefault()"
x-on:keydown.190="e.preventDefault()"
x-on:keydown...prevent
x-on:keydown.period.prevent
x-on:keydown.190.prevent
```

Any ideas?

**Comment by KevinBatdorf** [comment]
Alpine doesn't check the `keyCode` (a good PR probably if you want), so you'd have to do something like this:

```html
<input @keydown="if ($event.keyCode === 190) console.log('pressed')">
```
